DSharpPlus.Commands/Processors/TextCommands/TextCommandConfiguration.cs

```diff
@@ -10,11 +10,16 @@ public record TextCommandConfiguration
     /// The function to use to resolve prefixes for commands.
     /// </summary>
     /// <remarks>For dynamic prefix resolving, <see cref="IPrefixResolver"/> registered to the <see cref="DiscordClient"/>'s <see cref="IServiceProvider"/> should be preferred.</remarks>
-    public ResolvePrefixDelegateAsync PrefixResolver { get; init; } = new DefaultPrefixResolver(true,"!").ResolvePrefixAsync;
+    public ResolvePrefixDelegateAsync PrefixResolver { get; init; } = new DefaultPrefixResolver(true, "!").ResolvePrefixAsync;
     public TextArgumentSplicer TextArgumentSplicer { get; init; } = DefaultTextArgumentSplicer.Splice;
     public char[] QuoteCharacters { get; init; } = ['"', '\'', '«', '»', '‘', '“', '„', '‟'];
     public bool IgnoreBots { get; init; } = true;
 
+    /// <summary>
+    /// Disables the exception thrown when a command is not found.
+    /// </summary>
+    public bool DisableCommandNotFoundException { get; init; }

```

Comment:
do we want to default this to true

Reply:
Not particularly. I find this error message to be helpful when creating new commands and learning they didn't register.

Reply:
It's useful for debugging and for broken slashies (since the alternative is a vague Discord error) but IMO for text commands on a serious bot it's extremely annoying for end users. 

Having it as an option that's well-documented is enough for me to stop caring though, since opting out becomes so easy.

Reply:
One (potentially silly) idea off the top of my head could be to have it only show (for text commands at least) within the configured debug guild.

Reply:
Worth mentioning for clarification that this PR is for text commands specifically

Reply:
poll is looking like we want to default this to true ( https://canary.discord.com/channels/379378609942560770/379386730538860554/1267254677196636272 )

Reply:
If its to disable and defaulted to true, would it make more sense to flip it so that the option is to enable instead and defaults to false? Might make more intuitive sense for an opt-in feature.

Reply:
that would indeed make more sense, yes

Reply:
+1 on having no "command not found"s by default. Generally, this is bad practice and ridiculously annoying when you unexpectedly get this message in regular chat.

also +1 on flipping the property to `EnableCommandNotFoundException` and having it be off by default.